rust-toolchain.toml

```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "nightly-2023-08-07"
+channel = "nightly-2023-10-21"
```

Comment:
See -

* https://github.com/sgodwincs/dlv-list-rs/pull/19
* https://github.com/risingwavelabs/risingwave/pull/12992

Reply:
How about upgrade to the latest nightly :D

Reply:
I'm trying to minimize the change. We can try to upgrade it in another PR but this one only include the most necessary ones.